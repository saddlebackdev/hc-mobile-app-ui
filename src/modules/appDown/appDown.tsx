// Modules
import * as React from 'react';
import Styled from 'styled-components/native';
import {majorScale} from '../scales';

// Types
import {IProps} from './appDown.types';

// Shared
import Text from '../text/text';
import Heading from '../heading/heading';
import {SafeAreaView} from 'react-native';
import Button from '../button/button';

const StyledCard = Styled.View`
 justify-content: center;
  padding-horizontal: 45px;
  padding-vertical: 45px;
  width: 100%;
  height: 100%;
`;

const AnimationWrapper = Styled.View`
height: 280px;
margin-bottom: ${majorScale(2)}px;;
`;

const HeadingWrapper = Styled.View`
margin-bottom: ${majorScale(2)}px;;
`;

const TextWrapper = Styled.View`
margin-bottom: ${majorScale(3)}px;;
`;

const ButtonWrapper = Styled.View`
margin-bottom: ${majorScale(3)}px;;
`;

export const AppDown: React.FC<IProps> = ({
  onRestart,
  AnimationElement,
}): React.ReactElement => (
  <React.Fragment>
    <SafeAreaView>
      <StyledCard testID="app-Down-main">
        <AnimationWrapper>{AnimationElement}</AnimationWrapper>
        <HeadingWrapper>
          <Heading variant="h2" alignment="center">
            We are down for maintanence.
          </Heading>
        </HeadingWrapper>
        <TextWrapper>
          <Text variant="body1" alignment="center">
            We&apos;re busy making some improvements to the app. Please check
            back in a bit. Thanks for your patience!
          </Text>
        </TextWrapper>
        <ButtonWrapper>
          <Button testID="link" onPress={onRestart}>
            Restart Application
          </Button>
        </ButtonWrapper>
      </StyledCard>
    </SafeAreaView>
  </React.Fragment>
);
// Exports
export default AppDown;
