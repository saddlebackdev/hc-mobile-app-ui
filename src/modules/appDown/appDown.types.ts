// Item
export interface IProps {
  /** Function to be called when the item is pressed. */
  onRestart: any;

  animationElement?: JSX.Element;
}
