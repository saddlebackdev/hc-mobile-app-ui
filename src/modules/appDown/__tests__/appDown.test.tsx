// npx jest ./src/modules/appDown/__tests__/appDown.test.ts

import * as React from 'react';
import {fireEvent, render} from '@testing-library/react-native';

import {AppDown} from '../appDown';
import ThemeProvider from '../../theming/theme-provider';
import defaultTheme from '../../theming/default-theme';

const onPressMock = jest.fn();

describe('AppDown', () => {
  describe('Rendering', () => {
    it('renders itself correctly', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <AppDown onRestart={onPressMock} />
        </ThemeProvider>,
      );

      const list = wrapper.getByTestId('app-Down-main');
      expect(list).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Interactions', () => {
    it('should call onPress method when pressed', () => {
      const wrapper = render(
        <ThemeProvider theme={defaultTheme}>
          <AppDown onRestart={onPressMock} />
        </ThemeProvider>,
      );

      const moreItem = wrapper.getByTestId('link');
      expect(moreItem).toBeDefined();
      fireEvent.press(moreItem);
      expect(onPressMock).toHaveBeenCalledTimes(1);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
