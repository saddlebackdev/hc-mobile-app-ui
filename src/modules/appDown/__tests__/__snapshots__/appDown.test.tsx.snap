// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppDown Interactions should call onPress method when pressed 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Array [
        Object {
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID="app-Down-main"
  >
    <View
      style={
        Array [
          Object {
            "justifyContent": "center",
            "paddingHorizontal": 45,
            "paddingVertical": 45,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 280,
              "marginBottom": 22,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "marginBottom": 22,
            },
          ]
        }
      >
        <Text
          alignment="center"
          style={
            Array [
              Object {
                "color": "#1C2530",
                "fontFamily": "System",
                "fontSize": 20,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
          testID="heading"
          variant="h2"
        >
          We are down for maintanence.
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 33,
            },
          ]
        }
      >
        <Text
          alignment="center"
          style={
            Array [
              Object {
                "color": "#1C2530",
                "fontFamily": "System",
                "fontSize": 18,
                "fontStyle": "normal",
                "fontWeight": "400",
                "textAlign": "center",
              },
            ]
          }
          testID="text"
          variant="body1"
        >
          We're busy making some improvements to the app. Please check back in a bit. Thanks for your patience!
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 33,
            },
          ]
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#00AEFF",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 0,
              "display": "flex",
              "flexDirection": "row",
              "height": 44,
              "justifyContent": "center",
              "opacity": 1,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 0,
              "width": "100%",
            }
          }
          testID="link"
        >
          <Text
            appearance="filled"
            color="primary"
            disabled={false}
            small={false}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "600",
                },
              ]
            }
            testID="styled-label-link"
          >
            Restart Application
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`AppDown Rendering renders itself correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Array [
        Object {
          "height": "100%",
          "width": "100%",
        },
      ]
    }
    testID="app-Down-main"
  >
    <View
      style={
        Array [
          Object {
            "justifyContent": "center",
            "paddingHorizontal": 45,
            "paddingVertical": 45,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "height": 280,
              "marginBottom": 22,
            },
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "marginBottom": 22,
            },
          ]
        }
      >
        <Text
          alignment="center"
          style={
            Array [
              Object {
                "color": "#1C2530",
                "fontFamily": "System",
                "fontSize": 20,
                "fontWeight": "700",
                "textAlign": "center",
              },
            ]
          }
          testID="heading"
          variant="h2"
        >
          We are down for maintanence.
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 33,
            },
          ]
        }
      >
        <Text
          alignment="center"
          style={
            Array [
              Object {
                "color": "#1C2530",
                "fontFamily": "System",
                "fontSize": 18,
                "fontStyle": "normal",
                "fontWeight": "400",
                "textAlign": "center",
              },
            ]
          }
          testID="text"
          variant="body1"
        >
          We're busy making some improvements to the app. Please check back in a bit. Thanks for your patience!
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 33,
            },
          ]
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          nativeID="animatedComponent"
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#00AEFF",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderWidth": 0,
              "display": "flex",
              "flexDirection": "row",
              "height": 44,
              "justifyContent": "center",
              "opacity": 1,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 1,
              "shadowRadius": 0,
              "width": "100%",
            }
          }
          testID="link"
        >
          <Text
            appearance="filled"
            color="primary"
            disabled={false}
            small={false}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "600",
                },
              ]
            }
            testID="styled-label-link"
          >
            Restart Application
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
